test/ryhti_client/test_copy_plan.py

```diff
@@ -0,0 +1,130 @@
+from collections.abc import Callable
+
+import pytest
+from sqlalchemy.orm import Session
+
+from database import models
+from ryhti_client.database_client import DatabaseClient
+
+
+@pytest.fixture
+def database_client(rw_connection_string: str) -> DatabaseClient:
+    return DatabaseClient(rw_connection_string)
+
+
+@pytest.fixture
+def extra_data(plan_matter_instance: models.PlanMatter) -> dict:
+    return {"name": "test_plan", "plan_matter_id": plan_matter_instance.id}
+
+
+def test_copy_plan(
+    database_client: DatabaseClient,
+    session: Session,
+    complete_test_plan: models.Plan,
+    remove_plan: Callable[[str], None],
+):
+    """Copies complete_test_plan object and checks that data was copied correctly."""
+    original_plan_id = complete_test_plan.id
+    copied_plan_id = database_client.copy_plan(original_plan_id)
+    assert copied_plan_id is not None
+
+    copied_plan = session.get(models.Plan, copied_plan_id)
+    assert copied_plan is not None
+    assert copied_plan.id != original_plan_id
+
+    assert copied_plan.name == {"fin": "Test Plan 1"}
```

Comment:
Testiä voisi muuttaa niin, että tarkistaa että alkuperäisellä ja kopioidulla kohteella on samat arvot. Ehkä kaikkia perusarvoja ei tarvitse testata, vaan voidaan luottaa, että `clone_model` kopioi arvot kyllä oikein. Tosin `clone_model` metodille voisi kirjoittaa oman yksikkötestin, joka tarkistaa että se toimii kuten me oletetaan.
```suggestion
    assert copied_plan.name == original_plan.name
```

Tärkeintä tässä testissä olisi mun mielestä tarkitaa, relaatiot on duplikoitunut oikein eli, että duplikoiduille objekteille on duplikoitunut vastaavat lapsikohteet kuin alkuperäisellä objektilla.